src/libraries/System.Net.Ping/tests/FunctionalTests/UnixPingUtilityTests.cs

```diff
@@ -20,16 +20,20 @@ public class UnixPingUtilityTests
         private const int IcmpHeaderLengthInBytes = 8;
 
         [Theory]
-        [InlineData(0)]
-        [InlineData(1)]
-        [InlineData(50)]
-        [InlineData(1000)]
+        [InlineData(0, 1500)]
```

Comment:
Do you need so many timeout variations given they're all the same code path and you're not validating?
Is zero worth a test?

Reply:
it may be better to add timeout test instead of reusing PacketSizeIsRespected. Since we do not measure accuracy, the goal really is to verify that the ping does not break.

Reply:
Yes, I thought about that. For instance, I can ping nonexistent address such as `1.2.3.4`. But I'm worrying that some implementations of ping can recognize ICMP Type 3 response (Destination Unreachable) and exits too early.

Reply:
I think for verifying command line options, pinging loopback is fine.

Reply:
I suggest to add a test for zero - it's possible on one of the platforms that zero causes Ping to fail, in which case we should handle it up front. On my Linux machine, it seems to accept it, I assume it works like an infinite timeout.

Reply:
New test for unreachable endpoint can run infinitely due to zero timeout. I fixed that and added zero timeout for the test.